simd/src/Kokkos_SIMD_Common.hpp

```diff
@@ -26,6 +26,10 @@ namespace Kokkos {
 
 namespace Experimental {
 
+namespace simd_abi {
+class scalar;
+}
```

Comment:
Why this instead of having more specialized overloads in `<Kokkos_SIMD_Scalar.hpp>`?